A docs panel open and never closes.

### Summary

I think we should have a force close command for any open document pop-up.

![image](https://user-images.githubusercontent.com/11697220/196597570-883f3376-230d-499e-a5f8-779e41f285f7.png)


### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. `hx`

I expected this to happen: Close open docs when I open any other docs by <SPACE> + k.

Instead, this happened: As you can see on my screen it always opens I even close the buffer but it is still there. Now to solve this I have to close and reopen the helix.

### Helix log

```
 
2022-10-17T14:00:52.751 helix_lsp::transport [ERROR] err <- "  51: proc_macro::bridge::client::run_client::{{closure}}::{{closure}}::
2022-10-18T12:40:51.357 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1711, character: 0 }, end: Position { line: 1711, character: 0 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: unmatched `}`", related_information: None, tags: None, data: None }
2022-10-18T12:40:51.964 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1710, character: 6 }, end: Position { line: 1710, character: 6 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: expected BANG", related_information: None, tags: None, data: None }
2022-10-18T12:40:51.964 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1710, character: 12 }, end: Position { line: 1710, character: 12 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: expected SEMICOLON", related_information: None, tags: None, data: None }
2022-10-18T12:40:51.964 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1711, character: 0 }, end: Position { line: 1711, character: 0 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: unmatched `}`", related_information: None, tags: None, data: None }
2022-10-18T12:41:12.416 helix_term::application [WARN] unhandled window/showMessage: ShowMessageParams { typ: Error, message: "overly long loop turn: 141.290271ms" }
2022-10-18T12:48:08.069 helix_term::application [WARN] unhandled window/showMessage: ShowMessageParams { typ: Error, message: "overly long loop turn: 143.542743ms" }
2022-10-18T12:48:36.148 helix_term::application [WARN] unhandled window/showMessage: ShowMessageParams { typ: Error, message: "overly long loop turn: 172.39674ms" }
2022-10-18T12:50:09.860 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1670, character: 6 }, end: Position { line: 1670, character: 6 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: expected BANG", related_information: None, tags: None, data: None }
2022-10-18T12:50:09.860 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1670, character: 12 }, end: Position { line: 1670, character: 12 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: expected SEMICOLON", related_information: None, tags: None, data: None }
2022-10-18T12:50:09.860 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1671, character: 0 }, end: Position { line: 1671, character: 0 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: unmatched `}`", related_information: None, tags: None, data: None }
2022-10-18T12:50:31.612 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1666, character: 6 }, end: Position { line: 1666, character: 6 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: expected BANG", related_information: None, tags: None, data: None }
2022-10-18T12:50:31.612 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1666, character: 12 }, end: Position { line: 1666, character: 12 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: expected SEMICOLON", related_information: None, tags: None, data: None }
2022-10-18T12:50:31.612 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 1667, character: 0 }, end: Position { line: 1667, character: 0 } }, severity: Some(Error), code: Some(String("syntax-error")), code_description: Some(CodeDescription { href: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("rust-analyzer.github.io")), port: None, path: "/manual.html", query: None, fragment: Some("syntax-error") } }), source: Some("rust-analyzer"), message: "Syntax Error: unmatched `}`", related_information: None, tags: None, data: None }
2022-10-18T17:03:13.835 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
```

### Platform

Linux

### Terminal Emulator

zellij

### Helix Version

master

--- a/helix-term/src/commands/lsp.rs
+++ b/helix-term/src/commands/lsp.rs
@@ -9,7 +9,7 @@ use tui::text::{Span, Spans};
 use super::{align_view, push_jump, Align, Context, Editor, Open};
 
 use helix_core::{path, Selection};
-use helix_view::{apply_transaction, editor::Action, theme::Style};
+use helix_view::{apply_transaction, document::Mode, editor::Action, theme::Style};
 
 use crate::{
     compositor::{self, Compositor},
@@ -954,6 +954,14 @@ pub fn signature_help_impl(cx: &mut Context, invoked: SignatureHelpInvoked) {
                 || SignatureHelp::visible_popup(compositor).is_some()
                 || was_manually_invoked)
             {
+                return;
+            }
+
+            // If the signature help invocation is automatic, don't show it outside of Insert Mode:
+            // it very probably means the server was a little slow to respond and the user has
+            // already moved on to something else, making a signature help popup will just be an
+            // annoyance, see https://github.com/helix-editor/helix/issues/3112
+            if !was_manually_invoked && editor.mode != Mode::Insert {
                 return;
             }
 

